lib/api/api-fetch/response.js

```diff
@@ -1,141 +1,367 @@
 'use strict'
 
-const Headers = require('./headers')
-const { Blob } = require('buffer')
-const { STATUS_CODES } = require('http')
-const {
-  NotSupportedError
-} = require('../../core/errors')
+const { Headers, HeadersList, fill } = require('./headers')
+const { extractBody, cloneBody, mixinBody } = require('./body')
 const util = require('../../core/util')
-
+const { kEnumerableProperty } = util
+const { responseURL } = require('./util')
+const { redirectStatus, nullBodyStatus, forbiddenHeaderNames } = require('./constants')
+const assert = require('assert')
 const {
-  kType,
-  kStatus,
-  kStatusText,
-  kUrlList,
+  kState,
   kHeaders,
-  kBody
+  kGuard
 } = require('./symbols')
+const { kHeadersList } = require('../../core/symbols')
 
+// https://fetch.spec.whatwg.org/#response-class
 class Response {
-  constructor ({
-    type,
-    url,
-    body,
-    statusCode,
-    headers,
-    context
-  }) {
-    this[kType] = type || 'default'
-    this[kStatus] = statusCode || 0
-    this[kStatusText] = STATUS_CODES[statusCode] || ''
-    this[kUrlList] = Array.isArray(url) ? url : (url ? [url] : [])
-    this[kHeaders] = headers || new Headers()
-    this[kBody] = body || null
-
-    if (context && context.history) {
-      this[kUrlList].push(...context.history)
+  // Creates network error Response.
+  static error () {
+    // The static error() method steps are to return the result of creating a
+    // Response object, given a new network error, "immutable", and this’s
+    // relevant Realm.
+    // TODO: relevant Realm?
+    const responseObject = new Response()
+    responseObject[kState] = makeNetworkError()
+    responseObject[kHeaders][kHeadersList] = responseObject[kState].headersList
+    responseObject[kHeaders][kGuard] = 'immutable'
+    return responseObject
+  }
+
+  // Creates a redirect Response that redirects to url with status status.
+  static redirect (url, status = 302) {
+    // 1. Let parsedURL be the result of parsing url with current settings
+    // object’s API base URL.
+    // 2. If parsedURL is failure, then throw a TypeError.
+    // TODO: base-URL?
```

Comment:
Doesn't apply to us. Node.js doesn't run in a specific context where `baseURL` is applied. Unless this will be a feature?